### PR TITLE
[KernelGen][Nvidia] Add batch_norm_stats operator with Triton kernel

### PR DESCRIPTION
**Pull request by LoserCheems**
## Summary

Add `batch_norm_stats` operator implementation using a pure Triton kernel.

`torch.batch_norm_stats(input, eps)` computes per-channel mean and inverse standard deviation for batch normalization.

## Implementation Details

- **Single fused kernel**: Computes both mean and invstd in one dispatch
- **Grid design**: `(C,)` — one program per channel
- **Tiling**: 2D tiled reduction over batch and spatial dimensions (BLOCK_M × BLOCK_N)
- **Autotune**: Uses project-standard `batch_norm` autotune + heuristics configs
- **Multi-channel kernel**: For large-C small-spatial shapes (e.g. C≥256, spatial≤256), a specialized kernel processes multiple channels per program to reduce grid overhead
- **Precision**: float32 accumulator regardless of input dtype

### float32

| Shape | PyTorch (ms) | Triton (ms) | Speedup |
|-------|-------------|-------------|---------|
| (2, 64, 112, 112) | 0.0289 | 0.0114 | **2.52x** |
| (8, 128, 56, 56) | 0.0317 | 0.0115 | **2.76x** |
| (16, 256, 28, 28) | 0.0242 | 0.0126 | **1.91x** |
| (32, 512, 14, 14) | 0.0273 | 0.0168 | **1.63x** |
| (64, 1024, 7, 7) | 0.0318 | 0.0380 | 0.84x |
| (128, 2048, 4, 4) | 0.0558 | 0.0340 | **1.64x** |

**Average speedup: ~1.88x**

### Performance Note

The (128, 2048, 4, 4) shape was optimized with a multi-channel kernel that processes 8 channels per program, reducing the grid from 2048 to 256 blocks.

## Multi-backend Testing

| Backend | Accuracy | Speedup | Cases | Note |
|---------|----------|---------|-------|------|
| Nvidia H20 | PASS | ~1.88x | 6 | cudagraph mode |
| Tianshu | — | — | — | Not tested |
| Muxi | — | — | — | Not tested |
| Ascend | — | — | — | Not tested |
| Hygon | — | — | — | Not tested |

Tested-on: Nvidia H20

### CODE REVIEW

**Comment by bin913** [comment]
Thank you for your submission. The performance of this shape (128, 2048, 4, 4) is relatively poor. May I ask if it can be further improved

**Comment by LoserCheems** [comment]
Will look into optimizing (128, 2048, 4, 4). The small spatial dim (4x4=16) means each channel only reduces 2048 elements, which doesn't saturate the memory pipeline well. Will try tuning the block config for this regime.

**Comment by LoserCheems** [comment]
Optimized. Added a multi-channel kernel for large-C small-spatial shapes — each program processes multiple channels to reduce grid overhead. (128, 2048, 4, 4) now runs at 1.6-1.8x torch across all dtypes. Standard shapes unchanged.

File: tests/test_batch_norm_stats.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+)
+@pytest.mark.parametrize("dtype", utils.FLOAT_DTYPES)
+def test_batch_norm_stats(shape, dtype):
+    eps = 1e-5
+    inp = torch.randn(size=shape, dtype=dtype, device=flag_gems.device)
+
+    ref_inp = utils.to_reference(inp, True)
+    ref_mean, ref_invstd = torch.batch_norm_stats(ref_inp, eps)
+
+    res_mean, res_invstd = torch.batch_norm_stats(inp, eps)
```
Comment (bin913) [inline tests/test_batch_norm_stats.py:46]: Please ensure that the flags operator is called during testing

Comment (LoserCheems) [reply tests/test_batch_norm_stats.py]: done
